pkg/infra/tracing/tracing.go

```diff
@@ -136,6 +140,29 @@ func TraceIDFromContext(c context.Context, requireSampled bool) string {
 	return ""
 }
 
+// SpanFromContext returns the Span previously associated with ctx, or nil, if no such span could be found.
+// It is the equivalent of opentracing.SpanFromContext and trace.SpanFromContext.
+func SpanFromContext(ctx context.Context) Span {
+	// Look for both opentracing and opentelemetry spans.
+	if span := opentracing.SpanFromContext(ctx); span != nil {
+		return OpentracingSpan{span: span}
+	}
+	if span := trace.SpanFromContext(ctx); span != nil {
+		return OpentelemetrySpan{span: span}
+	}
+	return nil
+}
+
+// ContextWithSpan returns a new context.Context that holds a reference to the given span.
+// If span is nil, a new context without an active span is returned.
+// It is the equivalent of opentracing.ContextWithSpan and trace.ContextWithSpan.
+func ContextWithSpan(ctx context.Context, span Span) context.Context {
+	if span != nil {
+		return span.contextWithSpan(ctx)
```

Comment:
another approach here would be to just detect the span type and inject appropriately, avoiding need to extend the `Span` interface.

IMO the current approach is better decoupled, but curious what others think?